### PR TITLE
Update recipients for report sender

### PR DESCRIPTION
**Pull request by doncem**
Reference: original PR #597. Description below is copied

### JIRA link (if applicable) ###

[Automate sending reports via email](https://tools.hmcts.net/jira/browse/BPS-466)

### Change description ###

List of recipients must come through config which must be set from key vault. Since list is dynamic for painless config setup suggesting to store as comma separated value - usual approach for this kind of task.

Bonus: introducing extra config for auto configuration of `JavaMailSender` bean. ~Added `test-connection` flag. It tests connection before registering bean. Should fail the application failing that.~ Disabled at the moment. Giving the configuration is present, also _enabling_ possibility to automatically register report sender once `JavaMailSender` is enabled

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSenderDisabledTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+)
+@RunWith(SpringRunner.class)
+public class ReportSenderDisabledTest {
+
+    @Autowired
+    private ApplicationContext context;
+
+    @Test
+    public void should_not_have_report_sender_in_context() {
+        assertThat(context.getBeanNamesForType(ReportSender.class)).isEmpty();
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSenderDisabledTest.java]: This test is as such not required

Comment (doncem) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSenderDisabledTest.java]: proves the point `ReportSender` is not present when `JavaMailSender` is disabled. could even add more (if possible, not sure i tried) - check schedule is not registered and it would be pretty much functionally tested

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSender.java [hunk 2]
```diff
@@ -31,16 +35,11 @@
...
     ) {
         this.mailSender = mailSender;
         this.reportsService = reportsService;
-
-        if (recipients == null) {
-            this.recipients = new String[0];
-        } else {
-            this.recipients = Arrays.copyOf(recipients, recipients.length);
-        }
+        this.recipients = reportRecipientsCsv.split(",");
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSender.java]: does spring not support lists in config?
I'm pretty sure we've used it somewhere already. List of jurisdictions I think.

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSender.java]: it's the way you store list of recipients. since they have to be populated in vault and members of list vary this can be configured as comma separated values. especially when content is as we have now

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSender.java]: list of jurisdictions was stored the same way.
see here https://github.com/hmcts/bulk-scan-processor/pull/482

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -34,6 +34,20 @@ spring:
...
+    host: ${SMTP_HOST:false}
+    password: password
+    port: 587
+    properties:
+      mail:
+        smtp:
+          auth: true
+          starttls:
+            enable: true
+    username: username
```
Comment (rkondratowicz) [inline src/main/resources/application.yaml]: perhaps this could sit next to password.

File: src/main/resources/application.yaml [hunk 4]
```diff
@@ -34,6 +34,20 @@ spring:
           lob:
             # silence the 'wall-of-text' - unnecessary exception throw about blob types
             non_contextual_creation: true
+  mail:
+    host: ${SMTP_HOST:false}
+    password: password
```
Comment (rkondratowicz) [inline src/main/resources/application.yaml]: do we want this default value?

Comment (rkondratowicz) [reply src/main/resources/application.yaml]: any reason to not use explicit env var here?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSender.java (hunk 2 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSender.java]: i see what you mean now. automatically parse into array of strings as spring will do the split for you

File: src/main/resources/application.yaml (hunk 4 above)
Comment (doncem) [reply src/main/resources/application.yaml]: all env vars come in next pr when feature is actually enabled. all this is default/test setup

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSender.java [hunk 5]
```diff
@@ -31,16 +35,11 @@
...
     ) {
         this.mailSender = mailSender;
         this.reportsService = reportsService;
-
-        if (recipients == null) {
-            this.recipients = new String[0];
-        } else {
-            this.recipients = Arrays.copyOf(recipients, recipients.length);
-        }
+        this.recipients = reportRecipients;
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSender.java]: why this change??

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSenderTest.java [hunk 6]
```diff
@@ -65,19 +67,15 @@ public void should_send_email_to_all_recipients() throws Exception {
...
         // given
         ReportsService reportsService = mock(ReportsService.class);
         JavaMailSender mailSender = mock(JavaMailSender.class);
 
         given(mailSender.createMimeMessage())
             .willReturn(new JavaMailSenderImpl().createMimeMessage());
 
-        willThrow(MailSendException.class)
-            .given(mailSender)
-            .send(any(MimeMessage.class));
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/email/ReportSenderTest.java]: where's the exception that was supposed to be handled now?

File: src/main/resources/application.yaml (hunk 4 above)
Comment (lgonczar) [reply src/main/resources/application.yaml]: I think it's best for the app to crash if it doesn't receive the host. Otherwise we might have problems figuring out what's wrong if, for example, this env var is not set.
